lib/plug/cowboy/telemetry.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Plug.Cowboy.Telemetry do
+  def init(stream_id, req, opts) do
+    system_time = System.system_time()
+    start_time = System.monotonic_time()
+
+    :telemetry.execute([:plug_cowboy, :stream_handler, :start], %{system_time: system_time}, %{
```

Comment:
This name is a bit of an implementation detail, it's ok but what about:

```elixir
[:plug_cowboy, :connection, :start]
[:plug_cowboy, :request, :start]
[:cowboy, :connection, :start]
[:cowboy, :request, :start]
```

Reply:
We are thinking this may be a separate project altogether right? So I could go with `:cowboy`.

Reply:
Yes, but then we'd have to break the `[:plug_cowboy, :early_error]` event. Maybe that's OK?

Reply:
That’s ok. Let’s make the next release be v3.0.

Reply:
sounds good, i'll keep iterating on this and try rewriting it in erlang in this PR then we can extract it into a new package